agent/custom/general_navigation_action.py

```
@@ -624,306 +354,425 @@ def _iou(a, b):
+                end = (int(SCREEN_CENTER[0]), int(SCREEN_CENTER[1]))
+                mfaalog.info(f"[导航] 定向滑动 方向 ({direc[0]:.2f},{direc[1]:.2f}) 起点{start}->中心{end}")
+                controller.post_swipe(*start, *end, SWIPE_DURATION).wait()
                 time.sleep(1.5)
                 prevent_touch(context)   # 滑动后检查误触进入关卡
 
-            mfaalog.error(f"[导航] 滑到底未找到目标关卡: {target_quest} (共滑动 {swipe_count} 次)")
+            # 步骤4: 超轮未达目标 -> 失败(无兜底)
+            mfaalog.info("[导航] 方案B超最大轮数未达目标, 无法导航(无兜底)")
             return CustomAction.RunResult(success=False)
 
         except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)